usaon_vta_survey/util/full_sankey.py

```diff
@@ -0,0 +1,97 @@
+from itertools import chain
+
+from usaon_vta_survey.models.tables import (
+    Response,
+    ResponseApplication,
+    ResponseDataProduct,
+    ResponseObservingSystem,
+)
+
+
+# TODO: Can we do better than `object` here? Mypy doesn't correctly infer `str | int`
+def sankey(response: Response) -> list[list[object]]:
+    """Provide Sankey data structure, formatted for Highcharts."""
+    # Convert tuples to lists for passing to Javascript-land:
+    data = [list(e) for e in _sankey(response)]
+    return data
+
+
+def _sankey(response: Response) -> list[tuple[str, str, int]]:
+    """Provide a sankey data structure of applications, formatted for type checker."""
+    data1 = _applications_sankey(response)
+    data2 = _data_products_sankey(response)
+    data3 = _societal_benefit_areas_sankey(response)
+    data = data1 + data2 + data3
+    return data
+
+
+def _societal_benefit_areas_sankey(response: Response):
+    data = list(
+        chain(
+            *[
```

Comment:
It's not immediately obvious the structure of the returned list. Could you add a type annotation and/or docstring like the function below?

Reply:
Is this a more readable way to flatten the list? Just thinking out loud, no need to implement this.

```python
links = _application_societal_benefit_area_sankey_links(application)
data = [e for tpl in links for e in tpl]
```

Reply:
What about `chain.from_iterable`? Looks to do what we want out of the box :) https://docs.python.org/3/library/itertools.html#itertools.chain.from_iterable

Reply:
Using that :)